include/ginkgo/core/base/index_set.hpp

```diff
@@ -79,27 +79,50 @@ namespace gko {
  *
  * @tparam index_type  type of the indices being stored in the index set.
  *
- * @ingroup IndexSet
+ * @ingroup index_set
  */
 template <typename IndexType = int32>
-class IndexSet : public EnablePolymorphicObject<IndexSet<IndexType>> {
-    friend class EnablePolymorphicObject<IndexSet>;
-
+class index_set {
```

Comment:
Probably needs rule-of-five overloads, otherwise you can have an `index_set` that claims to contain data, but its arrays are empty.

Reply:
I think the `default` ones do the right thing here ? So rule of zero should be enough ?

Reply:
Ah, right. I see what you mean.